lib/miq_expression/tag.rb

```diff
@@ -36,7 +46,8 @@ def attribute_supported_by_sql?
   end
 
   def eql?(other)
-    other.try(:model) == model && other.try(:namespace) == namespace
+    other.try(:model) == model && other.try(:namespace) == namespace && other.try(:associations)
```

Comment:
hmm. (I know you didn't write this but...)
maybe `other && other.model == model && other.namespace == namespace && other.associations == associations`